chore!: Remove some use of [Export]

jonpryor opened the pull request:
Context: https://github.com/unoplatform/uno-private/issues/1422
Context: https://github.com/dotnet/maui/commit/70fbb835d74b7dd6268bfa51ba8f450d0bea22dd

While exploring the use of NativeAOT on Android for .NET 10, a sticking point is that `[Java.Interop.ExportAttribute]` does not "play nice" with NativeAOT; if a class contains an `[Export]` attributed-method, and that class is encountered during runtime, then an exception is thrown:

	F DOTNET  : FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: Methods such as __export__ are not implemented!
	F DOTNET  :    at Microsoft.Android.Runtime.ManagedTypeManager.RegisterNativeMembers(JniType, Type, ReadOnlySpan`1) + 0x386
	F DOTNET  :    at Java.Interop.ManagedPeer.RegisterNativeMembers(IntPtr jnienv, IntPtr klass, IntPtr n_nativeClass, IntPtr n_methods) + 0x195

This can be worked around for *instance* methods, but not for static methods.

Update `ApplicationActivity*.cs` and `NativeApplication.cs` to remove `[Export]`, and update any such attributed methods to throw a `NotSupportedException`.

Add a new `UnoWebViewHandlerJavascriptInterface.java` Java class, and update `UnoWebViewHandler.Android.cs` to implement `IUnoWebViewHandlerJavascriptInterface`, a'la dotnet/maui@70fbb835. This removes the use of `[Export]` from `UnoWebViewHandler.Android.cs`.

TODO: Review and update other uses of `[Export]`, such as in `UIElement.Android.cs`.

BREAKING CHANGE: the following methods now throw `NotSupportedException`:

  * `Microsoft.UI.Xaml.ApplicationActivity.GetTypeAssemblyFullName(string)` (Android renderer)
  * `Microsoft.UI.Xaml.ApplicationActivity.GetTypeAssemblyFullName(string)` (Android+Skia renderer)
  * `Microsoft.UI.Xaml.NativeApplication.GetTypeAssemblyFullName(string)`

**GitHub Issue:** closes #

<!-- Link to relevant GitHub issue if applicable. All PRs should be associated with an issue (GitHub issue or internal), unless the change is documentation related. -->

## PR Type:

<!--
Copy the labels that apply to this PR and paste them above:

- 🐞 Bugfix
- ✨ Feature
- 🎨 Code style update (formatting)
- 🔄 Refactoring (no functional changes, no api changes)
- 🏗️ Build or CI related changes
- 📚 Documentation content changes
- 🤖 Project automation
- 💬 Other... (Please describe)

-->


## What is the current behavior? 🤔

<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->


## What is the new behavior? 🚀

<!-- Please describe the new behavior after your modifications. -->

## PR Checklist ✅

Please check if your PR fulfills the following requirements:

- [ ] 📝 Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.
- [ ] 🧪 Added [Runtime tests, UI tests, or a manual test sample](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] 📚 Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] 🖼️ Validated PR `Screenshots Compare Test Run` results.
- [ ] ❗ Contains **NO** breaking changes

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below.
     Please note that breaking changes are likely to be rejected -->

## Other information ℹ️

<!-- Please provide any additional information if necessary -->

jonpryor committed:
chore!: Remove some use of [Export]

Context: https://github.com/unoplatform/uno-private/issues/1422
Context: https://github.com/dotnet/maui/commit/70fbb835d74b7dd6268bfa51ba8f450d0bea22dd

While exploring the use of NativeAOT on Android for .NET 10, a sticking
point is that `[Java.Interop.ExportAttribute]` does not "play nice"
with NativeAOT; if a class contains an `[Export]` attributed-method,
and that class is encountered during runtime, then an exception is thrown:

	F DOTNET  : FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: Methods such as __export__ are not implemented!
	F DOTNET  :    at Microsoft.Android.Runtime.ManagedTypeManager.RegisterNativeMembers(JniType, Type, ReadOnlySpan`1) + 0x386
	F DOTNET  :    at Java.Interop.ManagedPeer.RegisterNativeMembers(IntPtr jnienv, IntPtr klass, IntPtr n_nativeClass, IntPtr n_methods) + 0x195

This can be worked around for *instance* methods, but not for static
methods.

Update `ApplicationActivity*.cs` and `NativeApplication.cs` to
remove `[Export]`, and update any such attributed methods to throw
a `NotSupportedException`.

Add a new `UnoWebViewHandlerJavascriptInterface.java` Java class, and
update `UnoWebViewHandler.Android.cs` to implement
`IUnoWebViewHandlerJavascriptInterface`, a'la dotnet/maui@70fbb835.
This removes the use of `[Export]` from `UnoWebViewHandler.Android.cs`.

TODO: Review and update other uses of `[Export]`, such as in
`UIElement.Android.cs`.

BREAKING CHANGE: the following methods now throw `NotSupportedException`:

  * `Microsoft.UI.Xaml.ApplicationActivity.GetTypeAssemblyFullName(string)`
    (Android renderer)
  * `Microsoft.UI.Xaml.ApplicationActivity.GetTypeAssemblyFullName(string)`
    (Android+Skia renderer)
  * `Microsoft.UI.Xaml.NativeApplication.GetTypeAssemblyFullName(string)`

jonpryor commented:
Multiple implicit and explicit questions here:

1. *Will it build*? Does anything on CI call the `GetTypeAssemblyFullName()` methods which now throw `NotSupportedException`?
2. Is this change *acceptable*?
3. And/or should this change be *conditional* on .NET 10? i.e. .NET 9 builds work as before, while .NET 10 builds throw or just remove the method outright.

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21199/docs/index.html

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21199/wasm-skia-net9/index.html

jonpryor committed:
chore: fix build break

The `Uno.UI.Runtime.Skia.Android` build was failing with:

	C:\a\1\s\src\Uno.UI\UI\Xaml\Controls\WebView\Native\Android\UnoWebViewHandler.Android.cs(6,54): error CS0246: The type or namespace name 'IUnoWebViewHandlerJavascriptInterface' could not be found (are you missing a using directive or an assembly reference?)

Update `Uno.UI.Runtime.Skia.Android.csproj` to now reference
`Uno.UI.BindingHelper.Android.netcoremobile.csproj`, so that the
new `IUnoWebViewHandlerJavascriptInterface` type can be found.

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21199/wasm-skia-net9/index.html

jonpryor committed:
chore: fix src/SolutionTemplate/5.6/uno56droidioswasmskia build

Context: 8f7f7bc2b86aca49ac79737969426e6b457840de

**Tests - Templates > \* group_4 > Run Project Template Tests**
was failing:

	Uno56NugetLibrary.csproj : error NU1101: Unable to find package Uno.UI.BindingHelper.Android. No packages exist with this id in source(s): /agent/_work/1/s/.dotnet/library-packs, NuGet official package source, Solution Packages, uno-dev

The problem is that there is no `Uno.UI.BindingHelper.Android` NuGet
package, nor should there be.

The build was looking for one because commit 8f7f7bc2 added
`Uno.UI.BindingHelper.Android` as a `@(PackageReference)` to
`Uno.UI.Runtime.Skia.Android.csproj`, but neglected to set
[`%(TreatAsPackageReference)`][0]:

> Project to Project References Project to Project references will be,
> by default, be considered as nuget package references. However, this
> behavior can be overridden [by setting `%(TreatAsPackageReference)`=false]

Conveniently enough, this is *also* what most of the other
`@(PackageReference)`s do!

Update `Uno.UI.Runtime.Skia.Android.csproj` so that
`Uno.UI.BindingHelper.Android` is *not* treated as a NuGet package.

[0]: https://github.com/NuGet/Home/wiki/Adding-nuget-pack-as-a-msbuild-target/df5f292a271a5bdc7347e9e631be3625271d44bc

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21199/docs/index.html

unodevops commented:
⚠️⚠️ The build [172851](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=172851) has failed on Uno.UI - CI.

jonpryor committed:
chore: don't use Uno.UI.BindingHelper.Android for Java code

Context: 73b73b09cf3f783241f915d819c69e5801ea7103
Context: 8f7f7bc2b86aca49ac79737969426e6b457840de

`Uno.UI.Runtime.Skia.Android.csproj` explicitly adds
`UnoWebViewHandler.Android.cs` into `@(Compile)`, which means that
any changes to `UnoWebViewHandler.Android.cs` impacts the
`Uno.UI.Runtime.Skia.Android.csproj` build.

Thus, when 73b73b09 introduced new interface usage within
`UnoWebViewHandler.Android.cs`, this broke the build, as
`Uno.UI.Runtime.Skia.Android.csproj` didn't know about the new
`UnoWebViewHandlerJavascriptInterface` Java type.

Commit 8f7f7bc2 attempted to fix this by adding a `@(ProjectReference)`
to `Uno.UI.BindingHelper.Android` into `Uno.UI.Runtime.Skia.Android`.
This allowed things to build, but introduced a plethora of NuGet-
related errors (0cb3adb5) and now file errors:

	error XALNS7028: System.IO.FileNotFoundException:
	Could not load assembly 'Uno.UI.BindingHelper.Android, Version=255.255.255.255, Culture=neutral, PublicKeyToken='.
	Perhaps it doesn't exist in the Mono for Android profile?

Part of the conceptual problem here is that
`Uno.UI.Runtime.Skia.Android` uses types from
`src/Uno.UI` but not the entire infrastructure.  This is why adding
`Uno.UI.BindingHelper.Android` as an assembly reference failed: there's
too much "extra stuff" going on to prevent things from working.

Since the "cause" of the pain is that `Uno.UI.Runtime.Skia.Android`
is directly including files from `Uno.UI`, *continue* that path:
update `Uno.UI.Runtime.Skia.Android` to import
`UnoWebViewHandlerJavascriptInterface.java` as an
[`@(AndroidJavaSource)`][0].  This allows `Uno.UI.Runtime.Skia.Android`
to contain `IUnoWebViewHandlerJavascriptInterface` -- allowing it to
compile -- *without* using `Uno.UI.BindingHelper.Android`.

Alas, this requires updating `Uno.UI.Runtime.Skia.Android` to
add IDE0055 to `$(NoWarn)`, to ignore:

	src/Uno.UI.Runtime.Skia.Android/obj/**/generated/src/Uno.UI.IUnoWebViewHandlerJavascriptInterface.cs(6,17): error IDE0055: Fix formatting (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0055)
	src/Uno.UI.Runtime.Skia.Android/obj/**/generated/src/__NamespaceMapping__.cs(3,11): error IDE0055: Fix formatting (https://learn.microsoft.com/dotnet/fundamentals/code-analysis/style-rules/ide0055)

This also requires enabling nullability on
`UnoWebViewHandler.Android.cs`, addressing:

	src/Uno.UI/UI/Xaml/Controls/WebView/Native/Android/UnoWebViewHandler.Android.cs(16,14):
	error CS8767: Nullability of reference types in type of parameter 'message' of 'void UnoWebViewHandler.PostMessage(string message)'
	  doesn't match implicitly implemented member 'void IUnoWebViewHandlerJavascriptInterface.PostMessage(string? p0)'
	  (possibly because of nullability attributes).

[0]: https://learn.microsoft.com/en-us/dotnet/android/building-apps/build-items#androidjavasource

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21199/docs/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21199/docs/index.html

unodevops commented:
⚠️⚠️ The build [173133](https://dev.azure.com/uno-platform//Uno%20Platform/_build/results?buildId=173133) has failed on Uno.UI - CI.

jonpryor commented:
Rebased atop acd096ea1c5d26efa6462ea2a24cc83ff289e7bb, which makes for a known comparison point: https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=172962&view=results

acd096ea only shows test failures on **Tests - iOS Native** and **Publish - iOS Testflight**.  Nothing Android.

Compare to the current build, which has various Android-related failures: https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=173133&view=ms.vss-test-web.build-test-results-tab

jonpryor committed:
chore: Fix Given_WebView2.When_LocalFolder_File()

Context: 4d84ee31adb3e7ecd3fcbdc248b79fee78211d3e
Context: https://github.com/dotnet/java-interop/issues/1260
Context: https://github.com/dotnet/maui/commit/70fbb835d74b7dd6268bfa51ba8f450d0bea22dd
Context: https://github.com/dotnet/maui/commit/b1113d56866e4f41e0bb5b2be2d4c435a04630da

The `Given_WebView2.When_LocalFolder_File()` test was failing:

	Microsoft.VisualStudio.TestTools.UnitTesting.AssertFailedException: Timed out waiting for condition to be met. When_LocalFolder_File():344
	  at Private.Infrastructure.TestServices.WindowHelper.WaitFor(Func1 condition, Int32 timeoutMS, String message, String callerMemberName, Int32 lineNumber)
	  at Uno.UI.RuntimeTests.Tests.Microsoft_UI_Xaml_Controls.Given_WebView2.<When_LocalFolder_File>g__Do|7_0()
	  at Uno.UI.RuntimeTests.Helpers.TestHelper.RetryAssert(Func`1 assertion, Int32 count)
	  at Uno.UI.RuntimeTests.Tests.Microsoft_UI_Xaml_Controls.Given_WebView2.When_LocalFolder_File()

The timeout confused me for a bit: why would it be timing out?

`Given_WebView2.cs:344` provided more context:

	webView.NavigationCompleted += (sender, e) => navigated = true;
	…
	await TestServices.WindowHelper.WaitFor(() => navigated, 3000);

`TestServices.WindowHelper.WaitFor()` was timing out, because
`navigated` was never set, because the `webView.NavigationCompleted`
event was never raised.

`webView.NavigationCompleted`, meanwhile, is impacted by the removal
of `[Export]` on `UnoWebViewHandler.PostMessage()` in 4d84ee31!
Now things are making sense!

It turns out that dotnet/maui@70fbb835 did not work, because of
dotnet/java-interop#1260: the .NET for Android Java Callable Wrapper
generator does not process custom attributes on interface method
implementations.

Consequently, this:

	// Java
	public interface UnoWebViewHandlerJavascriptInterface {
	  void postMessage(String message);
	}

	// C#
	class UnoWebViewHandler : Java.Lang.Object, IUnoWebViewHandlerJavascriptInterface {
	  [JavascriptInterface]
	  public void PostMessage(string? message) => …
	}

***Did not work***, as the resulting Java Callable Wrapper for
`UnoWebViewHandler` did not have `@android.webkit.JavascriptInterface`.

The fix for MAUI was in dotnet/maui@b1113d56: use a *`class`*, not an
interface:

	// Java
	public abstract class UnoWebViewHandlerJavascriptInterface {
	  public abstract void postMessage(String message);
	}

	// C#
	class UnoWebViewHandler : UnoWebViewHandlerJavascriptInterface {
	  [JavascriptInterface]
	  public override void PostMessage(string? message) => …
	}

*This* causes the resulting Java Callable Wrapper to have the
necessary `@JavascriptInterface` annotation.

This in turn fixes the `Given_WebView2.When_LocalFolder_File()` test.

nventive-devops commented:
The [build 173206](https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=173206) found UI Test snapshots differences: `android-28-net8`: **23**, `android-28-net8-Snap`: **32**, `ios`: **3**, `ios-Snap`: **44**, `skia-linux-screenshots`: **89**, `skia-windows-screenshots`: **316**, `wasm`: **59**, `wasm-automated-net9.0-WinUI-Benchmarks-automated`: **0**, `wasm-automated-net9.0-WinUI-Default-automated`: **14**, `wasm-automated-net9.0-WinUI-RuntimeTests-0`: **0**, `wasm-automated-net9.0-WinUI-RuntimeTests-1`: **0**, `wasm-automated-net9.0-WinUI-RuntimeTests-2`: **0**

<details>
<summary>Details</summary>


* `android-28-net8`: **23** changed over 825

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `Detereminate_ProgressRing_Validation0_[#FF0000_#FF0000_#FF0000_#FF0000]_UITests_Microsoft_UI_Xaml_Controls_ProgressRing_WinUIDeterminateProgressRing`
  - `Detereminate_ProgressRing_Validation75_[#FF0000_#008000_#008000_#008000]_Progress-Ring-Value-75`
  - `Detereminate_ProgressRing_Validation50_[#FF0000_#008000_#008000_#FF0000]_Progress-Ring-Value-50`
  - `FlyoutTest_When_NoOverlayInputPassThroughElement_Then_DontPassThrough_woOff_UITests_Shared_Windows_UI_Xaml_Controls_Flyout_Flyout_OverlayInputPassThroughElement`
  - `NavigateBackAndForthBetweenMenuItemsAndSettings_Fluent_SamplesApp_Samples_Microsoft_UI_Xaml_Controls_NavigationViewTests_FluentStyle_FluentStyle_NavigationViewSample`
  - `SequentialAnimations_SamplesApp_Windows_UI_Xaml_Media_Animation_SequentialAnimationsPage`
  - `ListView_SelectedItems_SamplesApp_Windows_UI_Xaml_Controls_ListView_ListViewSelectedItems`
  - `ProgressRing_Visibility_Collapsed_UITests_Windows_UI_Xaml_Controls_ProgressRing_WindowsProgressRing_GH1220`
  - `UpDownEnabledTest_UITests_Shared_Microsoft_UI_Xaml_Controls_NumberBoxTests_NumberBoxPage`
  - `When_SingleSelectionWithItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `When_SingleSelectionWithoutItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `ProgressRing_IsEnabled_Running_UITests_Windows_UI_Xaml_Controls_ProgressRing_WindowsProgressRing_GH1220`
  - `WebView_NavigateToAnchor_click_anchor`
  - `WebView_NavigateToAnchor_Initial`
  - `WebView_NavigateToAnchor_navigate_to_anchor`
  - `When_Parent_PointerMoved_After_drag_on_ScrollViewer_-_touch`
  - `DecimalFormatterTest_UITests_Shared_Microsoft_UI_Xaml_Controls_NumberBoxTests_NumberBoxPage`
  - `When_MultipleSelectionWithoutItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `Detereminate_ProgressRing_Validation25_[#FF0000_#008000_#FF0000_#FF0000]_Progress-Ring-Value-25`
  - `When_Parent_PointerMoved_After_drag_on_non-scrolling_ScrollViewer`
  </details>

* `android-28-net8-Snap`: **32** changed over 1074

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `Border_Border_Corner_Overlap_UITests_Shared_Windows_UI_Xaml_Controls_BorderTests_Border_CornerOverlap`
  - `Microsoft_UI_Composition_UITests_Shared_Windows_UI_Composition_SKCanvasElement_Simple_UITests_Shared_Windows_UI_Composition_SKCanvasElement_Simple`
  - `Gesture_Recognizer_Pointer_Events_test_bench_UITests_Shared_Windows_UI_Input_GestureRecognizer_PointersEvents`
  - `NavigationView_MUXControlsTestApp_NavigationViewRS4Page_MUXControlsTestApp_NavigationViewRS4Page`
  - `TeachingTip_UITests_Microsoft_UI_Xaml_Controls_TeachingTipTests_TeachingTipPage_UITests_Microsoft_UI_Xaml_Controls_TeachingTipTests_TeachingTipPage`
  - `Border_Border_CornerRadius_Alignments_Uno_UI_Samples_UITests_BorderTestsControl_Border_CornerRadius_Alignments`
  - `Border_Uno_UI_Samples_UITests_BorderTestsControl_Border_BorderThickness_Uno_UI_Samples_UITests_BorderTestsControl_Border_BorderThickness`
  - `MUX_NumberBox_UITests_Shared_Microsoft_UI_Xaml_Controls_NumberBoxTests_NumberBoxPage`
  - `Image_UITests_Windows_UI_Xaml_Controls_ImageTests_SvgImageSource_Icons_UITests_Windows_UI_Xaml_Controls_ImageTests_SvgImageSource_Icons`
  - `ListView_ListView_With_ListViews_Count_Measure_UITests_Shared_Windows_UI_Xaml_Controls_ListView_ListView_With_ListViews_Count_Measure`
  - `NavigationView_MUXControlsTestApp_NavigationViewTopNavPage_MUXControlsTestApp_NavigationViewTopNavPage`
  - `SwipeControl_MUXControlsTestApp_SwipeControlPage2_MUXControlsTestApp_SwipeControlPage2`
  - `NavigationView_MUXControlsTestApp_NavigationViewCustomThemeResourcesPage_MUXControlsTestApp_NavigationViewCustomThemeResourcesPage`
  - `TextBlock_UITests_Shared_Windows_UI_Xaml_Controls_TextBlockControl_TextBlock_Layout_UITests_Shared_Windows_UI_Xaml_Controls_TextBlockControl_TextBlock_Layout`
  - `Transform_Basics_UITests_Shared_Windows_UI_Xaml_Media_Transform_Basics`
  - `Border_AutoBorderStretchwithbottommargin_Uno_UI_Samples_UITests_BorderTestsControl_AutoBorderStretchwithbottommargin`
  - `CommandBarFlyout_MUXControlsTestApp_CommandBarFlyoutPage_MUXControlsTestApp_CommandBarFlyoutPage`
  - `Pickers_UITests_Windows_UI_Xaml_Controls_CalendarView_CalendarView_Theming_UITests_Windows_UI_Xaml_Controls_CalendarView_CalendarView_Theming`
  - `MediaPlayerElement_Mini_player_UITests_Shared_Windows_UI_Xaml_Controls_MediaPlayerElement_MediaPlayerElement_Minimal`
  - `Brushes_ImageBrushChangingCornerRadius_Uno_UI_Samples_UITests_ImageBrushTestControl_ImageBrushChangingCornerRadius`
  </details>

* `ios`: **3** changed over 256

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `ProgressRing_Visibility_Collapsed_UITests_Windows_UI_Xaml_Controls_ProgressRing_WindowsProgressRing_GH1220`
  - `ProgressRing_IsEnabled_Running_UITests_Windows_UI_Xaml_Controls_ProgressRing_WindowsProgressRing_GH1220`
  - `Check_ListView_Swallows_Measure_UITests_Shared_Windows_UI_Xaml_Controls_ListView_ListView_With_ListViews_Count_Measure`
  </details>

* `ios-Snap`: **44** changed over 991

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `GridView_GenericApp_Views_Samples_Shared_Content_UITests_GridViewGrouped_GenericApp_Views_Samples_Shared_Content_UITests_GridViewGrouped`
  - `ListView_ListView_Aligned_Left_UITests_Shared_Windows_UI_Xaml_Controls_ListView_ListView_Aligned_Left`
  - `Default_Uno_UI_Samples_Content_UITests_WebView_WebView_AnchorNavigation_Uno_UI_Samples_Content_UITests_WebView_WebView_AnchorNavigation`
  - `Brushes_ImageBrushWithRotateTransform_Uno_UI_Samples_UITests_ImageBrushTestControl_ImageBrushWithRotateTransform`
  - `Brushes_PathImageBrushfill_Uno_UI_Samples_UITests_ImageBrushTestControl_PathImageBrushfill`
  - `ColorPicker_WinUIColorPickerPage_UITests_Shared_Microsoft_UI_Xaml_Controls_ColorPickerTests_WinUIColorPickerPage`
  - `Default_UITests_Windows_UI_Xaml_Controls_CommandBar_CommandBar_Native_Frame_UITests_Windows_UI_Xaml_Controls_CommandBar_CommandBar_Native_Frame`
  - `Icons_UITests_Shared_Windows_UI_Xaml_Controls_BitmapIconTests_BitmapIcon_Foreground_UITests_Shared_Windows_UI_Xaml_Controls_BitmapIconTests_BitmapIcon_Foreground`
  - `Image_UITests_Shared_Windows_UI_Xaml_Controls_ImageTests_Image_Stretch_Algmnt_Inf_Vertical_UITests_Shared_Windows_UI_Xaml_Controls_ImageTests_Image_Stretch_Algmnt_Inf_Vertical`
  - `Image_Uno_UI_Samples_UITests_Image_Image_Stretch_Alignment_Taller_Uno_UI_Samples_UITests_Image_Image_Stretch_Alignment_Taller`
  - `Image_UITests_Shared_Windows_UI_Xaml_Controls_ImageTests_Image_Stretch_Alignment_SizeOnControl_UITests_Shared_Windows_UI_Xaml_Controls_ImageTests_Image_Stretch_Alignment_SizeOnControl`
  - `Image_Uno_UI_Samples_UITests_Image_Image_Stretch_Alignment_Wider_Uno_UI_Samples_UITests_Image_Image_Stretch_Alignment_Wider`
  - `NavigationView_MUXControlsTestApp_NavigationViewMenuItemStretchPage_MUXControlsTestApp_NavigationViewMenuItemStretchPage`
  - `NavigationView_MUXControlsTestApp_NavigationViewMinimalPage_MUXControlsTestApp_NavigationViewMinimalPage`
  - `NavigationView_NavigationViewSample_SamplesApp_Samples_NavigationViewSample_NavigationViewSample`
  - `Pickers_UITests_Windows_UI_Xaml_Controls_CalendarView_CalendarView_Theming_UITests_Windows_UI_Xaml_Controls_CalendarView_CalendarView_Theming`
  - `Brushes_ImageBrushStretch_Uno_UI_Samples_UITests_ImageBrushTestControl_ImageBrushStretch`
  - `ContentControl_ContentControl_Nested_TemplatedParent_Uno_UI_Samples_Content_UITests_ContentControlTestsControl_ContentControl_Nested_TemplatedParent`
  - `Default_UITests_Shared_Windows_UI_Xaml_Controls_ListView_ListView_ObservableCollection_Unused_Space_UITests_Shared_Windows_UI_Xaml_Controls_ListView_ListView_ObservableCollection_Unused_Space`
  - `Icons_UITests_Microsoft_UI_Xaml_Controls_ImageIconTests_ImageIconPage_UITests_Microsoft_UI_Xaml_Controls_ImageIconTests_ImageIconPage`
  </details>

* `skia-linux-screenshots`: **89** changed over 2282

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `DropDownButtonPage.png-dark`
  - `DropDownButtonPage.png`
  - `Gamepad_Enumeration.png-dark`
  - `Gamepad_Enumeration.png`
  - `ImageBrushAlignmentXY2.png-dark`
  - `ImageBrushAlignmentXY2.png`
  - `DisplayInformation.png-dark`
  - `DisplayInformation.png`
  - `Buttons.png`
  - `Examples.png`
  - `ExpanderColorValidationPage.png-dark`
  - `ExpanderColorValidationPage.png`
  - `ImagesInlineInFlipView.png-dark`
  - `ImagesInlineInFlipView.png`
  - `ImageWithLateSourceUniformToFill.png-dark`
  - `ImageWithLateSourceUniformToFill.png`
  - `Buttons.png-dark`
  - `ClipboardTests.png-dark`
  - `ClipboardTests.png`
  - `ImageBrushInList.png-dark`
  </details>

* `skia-windows-screenshots`: **316** changed over 2282

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `AppBarButtonTest.png`
  - `AppBarButtonWithIconTest.png`
  - `AutoSizedTopLeft.png-dark`
  - `AutoSizedTopLeft.png`
  - `Examples.png`
  - `Grid_with_UserControl_VerticalAlignment_Variable_Height.png-dark`
  - `Grid_with_UserControl_VerticalAlignment_Variable_Height.png`
  - `Grid_with_UserControl_VerticalAlignment_Variable_Width.png-dark`
  - `Grid_with_UserControl_VerticalAlignment_Variable_Width.png`
  - `Arrange_Performance01.png-dark`
  - `AppBarButtonWithIconTest.png-dark`
  - `AutoSizedPathCentered.png-dark`
  - `Basics Pivot Test.png-dark`
  - `ButtonClippingTestsControl.png-dark`
  - `ButtonClippingTestsControl.png`
  - `CalendarView_Theming.png-dark`
  - `CalendarView_Theming.png`
  - `DoubleAnimationTestsControl.png-dark`
  - `DoubleAnimationTestsControl.png`
  - `DoubleAnimationUsingKeyFrames_TranslateX.png-dark`
  </details>

* `wasm`: **59** changed over 1056

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `MUXControlsTestApp.ExtraCommandBarFlyoutPage`
  - `UITests.Shared.Windows_UI.Xaml_Automation.AutomationProperties_AutomationId`
  - `UITests.Shared.Windows_UI_Composition.SKCanvasElement_Simple`
  - `UITests.Shared.Windows_UI_Xaml_Controls.ComboBox.ComboBox_ComboBoxItem_Selection`
  - `UITests.Windows_UI_Xaml_Controls.CommandBar.BackButtonImage.CommandBar_Frame`
  - `UITests.Windows_UI_Xaml_Controls.ImageTests.SvgImageSource_MultiLoad`
  - `Uno.UI.Samples.Content.UITests.ButtonTestsControl.Button_Enabled_Control_Disabled`
  - `Uno.UI.Samples.Content.UITests.ButtonTestsControl.CheckBox_Button`
  - `SamplesApp.Windows_UI_Xaml_Controls.ListView.ListView_Image`
  - `UITests.Microsoft_UI_Xaml_Controls.WebView2Tests.WebView2_NavigationProperties`
  - `UITests.Windows_UI_Xaml_Controls.CalendarView.CalendarView_Theming`
  - `SamplesApp.Windows_UI_Xaml_Controls.ListView.ListView_ItemContainerStyleSelector`
  - `UITests.Shared.Windows_UI_Xaml_Controls.ListView.ListView_ChangeView`
  - `UITests.Windows_UI_Xaml_Controls.BorderTests.PanelWithNullBrushAndNonZeroThickness`
  - `SamplesApp.Wasm.Windows_UI_Xaml_Controls.ComboBox.ComboBox_Corners`
  - `UITests.Shared.Windows_UI_Xaml_Media_Animation.SetTargetProperty`
  - `UITests.Windows_UI_Xaml_Controls.CommandBar.CommandBar_Native_With_TextBox`
  - `Uno.UI.Samples.Content.UITests.CommandBar.CommandBar_Padding`
  - `Uno.UI.Samples.Content.UITests.CommandBar.CommandBar_With_Long_Sentences`
  - `initial_state`
  </details>

* `wasm-automated-net9.0-WinUI-Benchmarks-automated`: **0** changed over 1

* `wasm-automated-net9.0-WinUI-Default-automated`: **14** changed over 877

  <details>
  <summary>🚨🚨 Comparison Details (first 20) 🚨🚨</summary>

  - `Default_StrokeThickness_MyPath`
  - `When_SingleSelectionWithItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `ListView_SelectedItems_SamplesApp_Windows_UI_Xaml_Controls_ListView_ListViewSelectedItems`
  - `When_NoSelectionWithItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `When_SingleSelectionWithoutItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  - `TestProgressRing_InitialState_UITests_Microsoft_UI_Xaml_Controls_ProgressRing_WinUIProgressRing_Features`
  - `Default_StrokeThickness_MyPolyline`
  - `Default_StrokeThickness_MyPolygon`
  - `SequentialAnimations_SamplesApp_Windows_UI_Xaml_Media_Animation_SequentialAnimationsPage`
  - `When_Theme_Changed_No_Crash_UITests_Windows_UI_Xaml_Controls_CalendarView_CalendarView_Theming`
  - `Default_StrokeThickness_MyEllipse`
  - `Default_StrokeThickness_MyRect`
  - `Default_StrokeThickness_MyLine`
  - `When_MultipleSelectionWithoutItemClick_Then_PointersEvents_UITests_Windows_UI_Xaml_Controls_ListView_ListView_Selection_Pointers`
  </details>

* `wasm-automated-net9.0-WinUI-RuntimeTests-0`: **0** changed over 1

* `wasm-automated-net9.0-WinUI-RuntimeTests-1`: **0** changed over 1

* `wasm-automated-net9.0-WinUI-RuntimeTests-2`: **0** changed over 1

</details>

jeromelaban commented on src/Uno.UI/UI/Xaml/NativeApplication.cs:
```diff
@@ -152,9 +152,9 @@ internal bool TryHandleIntent(Intent intent)
-		[Export(nameof(GetTypeAssemblyFullName))]
 		[global::System.ComponentModel.EditorBrowsable(global::System.ComponentModel.EditorBrowsableState.Never)]
-		public static string GetTypeAssemblyFullName(string type) => Type.GetType(type)?.Assembly.FullName;
+		public static string GetTypeAssemblyFullName(string type) =>
+			throw new NotSupportedException("`static` methods with [Export] are not supported on NativeAOT.");
```
This change will break this particular scenario:

https://github.com/unoplatform/Uno.UITest/blob/94d027295b779e28064aebf99aeaee2b393ad558/src/Uno.UITest.Helpers/Helpers/AppQueryExtensions.cs#L21

Uno itself is not using this `.XamlParent` helper for Uno.UITest, which is why it's likely working properly. Others maybe using it, but it's probably not possible to tell who's using it in the wild.

We could keep it for native renderers, and accept the fact that native renderers won't support native AOT. What's the build error when this attribute exists? Can it be ignored at app compiled time? If so, it may be good enough to keep it for native.

jonpryor replied:
I think that scenario is *already* broken, due to line 31: https://github.com/unoplatform/Uno.UITest/blob/94d027295b779e28064aebf99aeaee2b393ad558/src/Uno.UITest.Helpers/Helpers/AppQueryExtensions.cs#L31

```csharp
var controlType = "md5" + GetMd5Hash(nameSpace + ":" + assembly) + "." + typeName;
```

Use of MD5 for package names was removed in dotnet/java-interop@baf43b09, circa *2019*.  MD5 has not been supported since *Xamarin.Android 11.3.0*, before the .NET migration!

This code is *broken*, and has been for quite some time.

jonpryor replied:
The question remains around what to do about now.  The "safest" approach may be to restrict this to .NET 10 builds, a'la:

```csharp
#if !NET10_0_OR_GREATER
    [Export(nameof(GetTypeAssemblyFullName))]
#endif // !NET10_0_OR_GREATER
    [global::System.ComponentModel.EditorBrowsable(global::System.ComponentModel.EditorBrowsableState.Never)]
    public static string GetTypeAssemblyFullName(string type) =>
#if NET10_0_OR_GREATER
        throw new NotSupportedException("`static` methods with [Export] are not supported on NativeAOT.");
#else  // NET10_0_OR_GREATER
        Type.GetType(type)?.Assembly.FullName;
#endif // NET10_0_OR_GREATER
```

The *other* question is where this change belongs: separately (i.e. "this PR"), or as part of a .NET 10 PR?  The conditional change described above could live in PR.

jeromelaban replied:
Right, let's break for net10, that will be good enough, particularly it's been broken for a known codepath for a while :)

jonpryor committed:
chore: move breaking change to net10

Context: 4d84ee31adb3e7ecd3fcbdc248b79fee78211d3e

Commit 4d84ee31 noted a BREAKING CHANGE:

> BREAKING CHANGE: the following methods now throw `NotSupportedException`:
>
>   * `Microsoft.UI.Xaml.ApplicationActivity.GetTypeAssemblyFullName(string)`
>     (Android renderer)
>   * `Microsoft.UI.Xaml.ApplicationActivity.GetTypeAssemblyFullName(string)`
>     (Android+Skia renderer)
>   * `Microsoft.UI.Xaml.NativeApplication.GetTypeAssemblyFullName(string)`

*Delay* this breaking change so that it's part of the .NET 10 build,
not part of *all* supported builds.  This will ensure we don't break
existing .NET 9 semantics, while allowing .NET 10 to support NativeAOT.

unodevops commented:
🤖 Your WebAssembly Skia Sample App stage site is ready! Visit it here: https://unowasmprstaging.z20.web.core.windows.net/pr-21199/wasm-skia-net9/index.html

unodevops commented:
🤖 Your Docs stage site is ready! Visit it here: https://unodocsprstaging.z13.web.core.windows.net/pr-21199/docs/index.html
